Yoast/wordpress-seo
# LINGO-1403 sort alphabetically the terms in editing mode

## Context


* In editing mode, the terms added will be ordered based on the time they are added. While after saving, the terms will be rearranged alphabetically. This PR makes sure that we also order the terms are ordered alphabetically in editing mode.

## Summary


This PR can be summarized in the following changelog entry:

* Sorts terms order alphabetically in editing mode in Classic editor.

## Relevant technical choices:

* In this PR, I also created a separate file for creating a mock/fake document elements for testing purposes. This way we can reuse the data without remaking those elements every time we want to test something.
* I moved two general helper functions out of `dom.js` and create separate files for them.
* The terms are alphabetically ordered, and it's case-insensitive.

## Test instructions

### Test instructions for the acceptance test before the PR gets merged
This PR can be acceptance tested by following these steps:

Requirements
* Install and activate Classic editor
* Install and activate Yoast Test Helper
* Enable `Enable post types & taxonomies.` in Tools > Yoast Test > Post types & Taxonomies

**Test categories and custom categories**
**Note**: run the steps below in a regular post and in a custom post type, e.g. Book/Movie

   * Create a post/custom post type
   * Use category/custom category replacement variable in the snippet editor
      * E.g. if you use the Book custom post type (CPT), use `%%ct_book-category%%`, if you use the Movie CPT, use `%%ct_movie-category%%`. 
   * Create categories/custom categories in the "Categories" box inside the post, with this order: 
      * "Wild cats" (with capitalized "W")
         * Reasoning -> to make sure that initial capital letter doesn't affect the order
      * "domestic cats"
   * Make sure that the two categories/custom categories above are checked
   * Confirm that the categories/custom categories order displayed in the snippet editor is `domestic cats, Wild cats`
   * Save your changes
   * Confirm that the categories/custom categories are still displayed the same
   * Create a new category/custom category "Siamese cats" and also make sure that this category is checked
   * Confirm that the categories/custom categories order displayed is `domestic cats, Siamese cats, Wild cats`
   * Save your changes
   * Confirm that the categories/custom categories are still displayed the same

**Test tags and genres**
**Note**: run the steps below in a regular post and in a custom post type, e.g. Book/Movie

   * Create a post/custom post type or open an existing one
   * Use tag/genre replacement variable in the snippet editor
      * E.g. if you use the Book custom post type (CPT), use `%%ct_book-genre%%`, if you use the Movie CPT, use `%%ct_movie-genre%%`.
   * Add tags/genres in this order:
      * "cats"
      * "Dogs"
      * "bears"
   * Confirm that the tags/genres order displayed in the snippet editor is `bears, cats, Dogs`
   * Save your changes
   * Confirm that the tags/genres are still displayed the same
   * Add a new tag/genre "birds"
   * Confirm that the tags/genres order displayed in the snippet editor is `bears, birds, cats, Dogs`
   * Save your changes
   * Confirm that the tags/genres are still displayed the same

### Test instructions for QA when the code is in the RC




QA can test this PR by following these steps:

*

## Impact check

This PR affects the following parts of the plugin, which may require extra testing:

*

## UI changes


## Other environments


## Documentation


## Quality assurance


Fixes https://yoast.atlassian.net/browse/LINGO-1403
